shell/platform/android/io/flutter/view/AccessibilityViewEmbedder.java

```diff
@@ -470,32 +496,102 @@ private Long getSourceNodeId(@NonNull AccessibilityNodeInfo node) {
+                try {
+                    return (long) longArrayGetIndex.invoke(childNodesField.get(node), child);
+                // Using identical separate catch blocks to comply with the following lint:
+                // Error: Multi-catch with these reflection exceptions requires API level 19
+                // (current min is 16) because they get compiled to the common but new super
+                // type ReflectiveOperationException. As a workaround either create individual
+                // catch statements, or catch Exception. [NewApi]
+                } catch (IllegalAccessException e) {
+                    Log.w(TAG, e);
+                } catch (InvocationTargetException | ArrayIndexOutOfBoundsException e) {
+                    Log.w(TAG, e);
+                }
             }
             return null;
         }
 
         @Nullable
         private Long getParentNodeId(@NonNull AccessibilityNodeInfo node) {
-            if (getParentNodeId == null) {
+            if (getParentNodeId != null) {
+                try {
+                    return (long) getParentNodeId.invoke(node);
+                // Using identical separate catch blocks to comply with the following lint:
+                // Error: Multi-catch with these reflection exceptions requires API level 19
+                // (current min is 16) because they get compiled to the common but new super
+                // type ReflectiveOperationException. As a workaround either create individual
+                // catch statements, or catch Exception. [NewApi]
+                } catch (IllegalAccessException e) {
+                    Log.w(TAG, e);
+                } catch (InvocationTargetException e) {
+                    Log.w(TAG, e);
+                }
+            }
+
+            // Fall back on reading the ID from a serialized data if we absolutely have to.
+            return yoinkParentIdFromParcel(node);
+        }
+
+        @Nullable
+        private static Long yoinkParentIdFromParcel(AccessibilityNodeInfo node) {
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+                Log.w(TAG, "Unexpected Android version. Unable to find the parent ID.");
                 return null;
             }
-            try {
-                return (long) getParentNodeId.invoke(node);
-            } catch (IllegalAccessException e) {
-                Log.w(TAG, e);
-            } catch (InvocationTargetException e) {
-                Log.w(TAG, e);
+
+            // If this looks like it's failing, that's because it probably is. This method is
+            // relying on the implementation details of `AccessibilityNodeInfo#writeToParcel` in
+            // order to find the particular bit in the opaque parcel that represents mParentNodeId.
+            // If these implementation details change from our assumptions in this method, this will
+            // silently break.
+            AccessibilityNodeInfo copy = AccessibilityNodeInfo.obtain(node);
```

Comment:
nit: it wasn't immediately obvious to me why we need a copy, let's mention in a comment that writeToParcel recycles the object.